src/retryable.rs

```diff
@@ -0,0 +1,128 @@
+use std::{future::Future, time::Duration};
+
+#[cfg(feature = "async")]
+use async_std::task::sleep;
+#[cfg(feature = "async")]
+use reqwest::{RequestBuilder, Response as ReqwestResponse, StatusCode};
+
+#[cfg(feature = "blocking")]
+use ureq::{Error as UReqError, Request, Response as UReqResponse};
+
+const SECOND: u64 = 1_000;
+const BASE_BACKOFF_MS: u64 = 5_000;
+const DEFAULT_MAX_RETRIES: u32 = 5;
+
+pub const RETRY_TOO_MANY_REQUEST_ONLY: [StatusCode; 1] = [StatusCode::TOO_MANY_REQUESTS];
+
+pub trait AsyncRetryable<R, E = crate::Error> {
+    fn exec_with_retry(
+        self,
+        retryable_error_codes: Vec<StatusCode>,
+        max_retries: Option<u32>,
+    ) -> impl Future<Output = Result<R, E>>;
+}
+
+pub trait SyncRetryable<R, E = crate::Error> {
+    fn exec_with_retry(
+        self,
+        retryable_error_codes: Vec<StatusCode>,
+        max_retries: Option<u32>,
+    ) -> Result<R, E>;
+}
+
+fn compute_backoff(retry_after: Option<&str>, backoff: Duration) -> Duration {
+    match retry_after {
+        // If response has a Retry-After header, parse it and use it as backoff
+        Some(retry_after) => {
+            return Duration::from_millis(
+                retry_after
+                    .parse::<u64>()
```

Comment:
We should be prepared to parse a date as well, according to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After

Reply:
Done in [1c64b9b](https://github.com/bitcoindevkit/rust-esplora-client/pull/71/commits/1c64b9bdfe0d3cc17c6630db2dfdb1236b9b11ae)